### ISSUE TITLE
CA1508 (Avoid Dead Conditional Code) using statement continues to give false positive 

### ISSUE THREAD

**Issue by abatishchev** [dotnet/roslyn-analyzers#3063]
The continuation of #2250 as the latest version still has this (or similar) issue.

Would be great if you could prioritize the resolution as an internal partner's ask, Identity Platform Engineering would be grateful.

#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.7 (Latest)

#### Diagnostic ID

CA1508 (no article at https://docs.microsoft.com/en-us/visualstudio/code-quality/)

#### Repro steps

```c#
using (var fabricClient = new FabricClient())
{
}
```

#### Expected behavior

No error

#### Actual behavior

>'fabricClient = new FabricClient()' is never 'null'

More examples:

![image](https://user-images.githubusercontent.com/351644/69371638-17d7e600-0c55-11ea-8b62-721ef03ac9a7.png)

**Comment by mavasani** [comment]
Would need to add a check for `!operation.IsImplicit` here: https://github.com/dotnet/roslyn-analyzers/blob/6d265703c1ccd10176dc77e768412cf1422f9153/src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/AvoidDeadConditionalCode.cs#L69-L75

**Comment by RobSiklos** [comment]
Another repro, not sure if it's the same problem:

```csharp
private static void Foo(string values)
{
    // If you comment out the following line, the warning goes away.
    if (values == null) { }

    // The following line of code generates the warning:
    // CA1508 'sw = new StringWriter()' is never 'null'. Remove or refactor the condition(s) to avoid dead code.
    using (var sw = new StringWriter()) { }
}
```

**Comment by PeteBoyRocket** [comment]
@mavasani Am I reading this test correctly here: https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidDeadConditionalCode_NullAnalysis.cs#L5454

I think this covers this case? If it does, do that mean this issue is fixed and will be in the next release?
